shared/KOSocket.cpp

```diff
@@ -25,6 +25,8 @@ void KOSocket::OnConnect()
 void KOSocket::OnRead() 
 {
 	Packet pkt;
+	std::recursive_mutex m_lock;
+	Guard Lock(m_lock);
```

Comment:
```suggestion
```
This doesn't seem to be helpful to `OnRead` function. Reason; `SocketWorkerThread` runs on its own single thread, so there is no point in adding thread-safe mechanisms unless there are multiple threads executing the same function.